### PR TITLE
feat: add structured timing logs for all proxy calls

### PR DESCRIPTION
**Pull request by chris-dickson**
This adds an annotation for recording structured timing logs for all proxy calls in terarium. 

**NOTE: This is a bit annoying/manual but was unfortunately the cleanest way I could get this to work in Quarkus.**

Example usage:
```java
package software.uncharted.terarium.hmiserver.proxies.myservice;

@RegisterRestClient(configKey = "test")
@Path("/hello")
public interface ExternalPerformerProxy {
	
	@Path("/world")
        @POST
	@LogRestClientTime(group = "external-performer-services")
	Response foo(String bar);
}
```

Would output
```
PROXY_REQUEST | adam | foo | software.uncharted.terarium.hmiserver.proxies.myservice.ExternalPerformerProxy | /hello/world | POST | external-performer-services | 104
```

We log:
- `PROXY_REQUEST` identifier
- The username of the user making the proxy, `Anonymous` if it is done on behalf of the system itself
- The name of the proxy method (in Java)
- The name of the parent class (the actual rest client)
- The rest path being executed
- The method type (`GET`, `POST`, etc)
- An optional `group` that can be specified on the annotation, if not provided, defaults to the package name for grouping
- The time in ms the call took

Example of output:
```
2023-07-13 10:29:52,656 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-0) PROXY_REQUEST | adam | getProjects | software.uncharted.terarium.hmiserver.proxies.dataservice.ProjectProxy | /projects | GET | dataservice | 45
2023-07-13 10:29:53,084 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-0) PROXY_REQUEST | adam | getAssets | software.uncharted.terarium.hmiserver.proxies.dataservice.ProjectProxy | /projects/{project_id}/assets | GET | dataservice | 428
2023-07-13 10:29:53,147 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-0) PROXY_REQUEST | adam | getAssets | software.uncharted.terarium.hmiserver.proxies.dataservice.ProjectProxy | /projects/{project_id}/assets | GET | dataservice | 62
2023-07-13 10:29:53,644 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-0) PROXY_REQUEST | adam | getDocuments | software.uncharted.terarium.hmiserver.proxies.documentservice.DocumentProxy | /api/v2/articles | GET | documentservice | 419
2023-07-13 10:29:54,412 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-1) PROXY_REQUEST | adam | getDocuments | software.uncharted.terarium.hmiserver.proxies.documentservice.DocumentProxy | /api/v2/articles | GET | documentservice | 1813
2023-07-13 10:29:54,541 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-1) PROXY_REQUEST | adam | searchConceptsUsingFacets | software.uncharted.terarium.hmiserver.proxies.dataservice.ConceptProxy | /concepts/facets | GET | dataservice | 6
2023-07-13 10:30:14,356 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-1) PROXY_REQUEST | adam | getProjects | software.uncharted.terarium.hmiserver.proxies.dataservice.ProjectProxy | /projects | GET | dataservice | 5
2023-07-13 10:30:14,358 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-0) PROXY_REQUEST | adam | getProject | software.uncharted.terarium.hmiserver.proxies.dataservice.ProjectProxy | /projects/{id} | GET | dataservice | 6
2023-07-13 10:30:14,683 INFO  [sof.unc.ter.hmi.ser.StructuredLog] (executor-thread-0) PROXY_REQUEST | adam | getAssets | software.uncharted.terarium.hmiserver.proxies.dataservice.ProjectProxy | /projects/{project_id}/assets | GET | dataservice | 301
```

### CODE REVIEW

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/proxies/dataservice/ArtifactProxy.java [hunk 1]
```diff
@@ -19,30 +20,36 @@
 public interface ArtifactProxy {
 
 	@GET
+	@LogRestClientTime
 	List<Artifact> getArtifacts( @DefaultValue("100") @QueryParam("page_size") Integer pageSize,
 															 @DefaultValue("0") @QueryParam("page") Integer page);
 
 	@POST
+	@LogRestClientTime
```
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/proxies/dataservice/ArtifactProxy.java:28]: All of these annotations are using the default grouping on the log i think - is that intentional? Should we define different groups?

Comment (YohannParis) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/proxies/dataservice/ArtifactProxy.java:28]: I think those could be define in another PR after the hackathon or if MITRE requires some specifics grouping.

**Review by YohannParis — APPROVED** [review]

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/proxies/dataservice/ArtifactProxy.java (hunk 1 above)
Comment (chris-dickson) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/proxies/dataservice/ArtifactProxy.java:28]: Agreed.  Someone that knows the systems better than I do should go in and label them

Comment (dvince2) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/proxies/dataservice/ArtifactProxy.java:28]: Sounds good!

**Pull request merged**